### PR TITLE
Fix error on getting diff of the root commit

### PR DESCRIPTION
**Pull request by itchyny**
Hi, I notices while `git rebase -i --root`, the plugin fails to show the diff of the root commit.
```
fatal: ambiguous argument 'XXX~1': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```
I fixed the error by falling back to show the root commit diff.

### CODE REVIEW

File: autoload/auto_git_diff.vim [hunk 1]
```diff
@@ -70,6 +68,25 @@ function! s:show_git_diff_impl(hash, vertsplit, opts) abort
     silent wincmd p
 endfunction
 
+function! s:get_git_diff(hash, opts) abort
+    let prefix = has("win32") ? "set LANG=C & " : "LANG=C "
+
+    let diff_command = "git diff ".a:opts." ".a:hash."~1 ".a:hash
+    silent let out = system(prefix.diff_command)
```
Comment (hotwatermorning) [inline autoload/auto_git_diff.vim:75]: It seems that nothing will be shown if `git diff` command fails here for some reason except for the first commit.

Please save the contents of `out` and return it if the next execution of `git diff` is failed.

**Comment by hotwatermorning** [comment]
Thank so much for sending PR.
I didn't noticed the problem with the `--root` option.

I checked your changes and add a comment. Please confirm it.

File: autoload/auto_git_diff.vim (hunk 1 above)
Comment (itchyny) [reply autoload/auto_git_diff.vim:75]: Right, fixed in 304176d.

**Comment by hotwatermorning** [comment]
Everything is alright. I merged.

Thank you for your contribution.

**Pull request merged**